### ISSUE TITLE
Make parameters into independent sensors

### ISSUE THREAD

**Issue by Chaoscontrol** [jesserockz/ha-leafspy#29]
**Describe the solution you'd like**
I have seen that most parameters from LeafSpy are recorded as attributes within the `device_tracker` sensor. While I appreciate the info there, I think they would make much more sense to be independent sensors themselves, so they get history tracked and they still belong to the same "Home Assistant Device". This would make them easier to use. 

**Describe alternatives you've considered**
Create template sensors that return the attribute value. 

**Additional context**
Attributes are not saved in the sensor's history, so you cannot keep a record of your odometer, or your SOH for example if they are attributes. Them being sensors I think it's the way Home Assistant expects for data like this. 

I just found this integration exists and I'm SO grateful for this. Carwings API has been deprecated in the UK, and this now allows me to track everything and even more from HA. Thank you SO much for the incredible work.

**Comment by wtadler** [comment]
I agree this would be great to add, but I personally haven't had time to do it! I would gladly test (and merge) a PR if you or anyone else would like to take a crack at it.

Glad you find the package useful. @jesserockz gets the vast majority of the credit, even though he's no longer actively maintaining it.

**Comment by Chaoscontrol** [comment]
I wish I knew how to dev it myself to create a PR, but honestly I'm clueless. 

All I can add is that it's pretty easy to create the template sensors. I did these for myself. 

![image](https://github.com/user-attachments/assets/690d66fc-49a5-4592-9d85-87fa40804f2c)
![image](https://github.com/user-attachments/assets/f78f3328-8804-4885-aa21-da735152b988)
![image](https://github.com/user-attachments/assets/6482ed97-a3a6-413e-8898-a0b17362197b)
![image](https://github.com/user-attachments/assets/440cd87a-8c33-416b-bb73-c1e81a48971d)

```
{{ state_attr('device_tracker.leaf', 'battery_level') }}
{{ state_attr('device_tracker.leaf', 'battery_health') }}
{{ state_attr('device_tracker.leaf', 'odometer') }}
```
The hard part is honestly done already, which is finding how to extract those from LeafSpy. The mod would just be creating different sensors instead of adding those elements as attributes. They all share the same structure too (`sensor`).

**Comment by wtadler** [comment]
Yeah, I'm using those template sensors too. Including sensors properly in the package is a little more complicated than just adding templates, but it should still be fairly straightforward. Hopefully I'll find some time to do it.

**Comment by Chaoscontrol** [comment]
I've been fighting with Claude and ChatGPT for them to help me creating the new sensor, but I haven't been able to make it show up. I'm not familiar enough to create PRs tbh. 

Could you first have a quick look at the `sensor.py` and see if it looks kinda decent? I've tried with SOH. 
```python
"""Sensor platform that adds support for Leaf Spy."""
import logging
from homeassistant.components.sensor import SensorEntity, SensorStateClass
from homeassistant.const import PERCENTAGE
from homeassistant.util import slugify
from .const import DOMAIN as LS_DOMAIN
from homeassistant.helpers.dispatcher import async_dispatcher_connect

_LOGGER = logging.getLogger(__name__)

async def async_setup_entry(hass, entry, async_add_entities):
    """Set up Leaf Spy sensors based on a config entry."""
    if 'sensors' not in hass.data[LS_DOMAIN]:
        hass.data[LS_DOMAIN]['sensors'] = {}

    async def _process_message(context, message):
        """Process the message."""
        dev_id = slugify(f'leaf_{message["VIN"]}')
        
        entity = hass.data[LS_DOMAIN]['sensors'].get(dev_id)
        if entity is not None:
            entity.update_state(float(message['SOH']))
            return

        entity = LeafSpyBatteryHealthSensor(dev_id, float(message['SOH']))
        hass.data[LS_DOMAIN]['sensors'][dev_id] = entity
        async_add_entities([entity])

    async_dispatcher_connect(hass, LS_DOMAIN, _process_message)
    return True

class LeafSpyBatteryHealthSensor(SensorEntity):
    """Representation of the Battery Health (SOH) sensor."""

    def __init__(self, device_id, soh):
        """Initialize the sensor."""
        self._device_id = device_id
        self._soh = soh
        self._attr_state_class = SensorStateClass.TOTAL
        self._attr_native_unit_of_measurement = PERCENTAGE

    @property
    def unique_id(self):
        """Return a unique ID for the sensor."""
        return f"{self._device_id}_battery_health"

    @property
    def name(self):
        """Return the name of the sensor."""
        return f"Leaf Battery Health"

    @property
    def native_value(self):
        """Return the state of the sensor."""
        return self._soh

    @property
    def device_info(self):
        """Return device information."""
        return {
            "identifiers": {(LS_DOMAIN, self._device_id)},
        }

    def update_state(self, new_soh):
        """Update the sensor state."""
        self._soh = new_soh
        self.async_write_ha_state()
```

The only other change I did to support this was on `__init__.py` to add the new sensor platform: 

```
PLATFORMS = ["device_tracker", "sensor"]
```
Tried manually adding the changes to my HA and restarting, but it didn't show up, even after few iterations of Claude trying to make it more similar to the `device_tracker`. My guess is that it's not actually the sensor that it's wrong, but whatever is needed to make it show up. No errors in log or anything. 

Thanks!

**Comment by wtadler** [comment]
Thanks for giving this a shot! Have you tried sharing with Claude any sample code to try to mimic and adapt this repository to? I would try to find existing HA core integrations that are roughly similar to this one in terms of scope (i.e., small) and in terms of how it receives information.

**Comment by Chaoscontrol** [comment]
Yes, definitely. I gave him the `device_tracker.py` as example, and then also the init and the const. While iterating it kept trying to make it resemble the device tracker more and more in an attempt to make it appear,  but it never did. 

My HA server just died, so won't be able to try much for some days.

**Comment by Chaoscontrol** [comment]
F'ing hell, I'm stupid!

After a couple days I checked again and the new sensor has actually appeared! It must be that HA won't show a new sensor until it receives some value, which I wasn't forcing since I wasn't getting into the car. 

It might be that previous iterations of the code were already working lol. So now when I get the chance I'll try to define a default icon, and then to add some of the other sensors as well. :D Excited that I (AI) made it work!

![image](https://github.com/user-attachments/assets/2a7e6432-d7bb-4fe5-a5cc-74d7c8de5599)

**Comment by wtadler** [comment]
Very cool! Nice work. Let me know when you'd like me to test.

**Comment by Chaoscontrol** [comment]
I've managed to get all 3 sensors working. But then I've spent like 5h wrestling with the shit to make it so they retain their values over HA restart, with 0 success. They have their history, but the entities are not provided by the integration until a new message is received. But the `device_tracker` DOES retain its state somehow. 

I have asked over 5h to ghatGPT to mimic whatever the `device_tracker` is doing, but it keeps doing shit and still not working.

But I'm noticing the `device_tracker` seems to erase its full history on every HA restart, but somehow it retains its prev values for all its attributes too. Is this the case? Can you confirm this with the default integration, or have I messed up something?

It's driving me nuts.

**Comment by wtadler** [comment]
I just restarted HA and I still have my `device_tracker` history and attribute values.

**Comment by wtadler** [comment]
I managed to do a test and it works! I'm looking forward to trying it out again when you add the other sensors. Make sure to include a device class. That will take care of icons and such, for many sensors.

**Comment by Chaoscontrol** [comment]
I don't even remember what was wrong with that version. I've gone through SO many iterations at this point. 
Adding the other 2 sensors is easy really. It's just that I wanted to deal with restoring states after HA restart, but when I achieved it it caused so many other issues. 

Might go back to this version that I shared and check again from there.

**Comment by wtadler** [comment]
Closed by PR #30 (merged into the sensor_entities branch), but there is more work to do still. Adding some issues to note those.
